### PR TITLE
Decrease Pylint Violation For Sans and Reduction GUI files

### LINKED ISSUE #15170: decrease the number of pylint violations for all SANS & reduction_gui files
Four of the files from top 15 pylint offenders are under the following directory: `scripts/SANS/` and 
`scripts/Interface/reduction_gui`, adding up to total of 186 violations. 
These files include:
- SANS/ISISCommandInterface.py
- reduction_gui/widgets/sans/stitcher.py
- reduction_gui/reduction/scripter.py
- scripts/SANS/isis_reducer.py
- reduction/instruments/sans/sans_reduction_steps.py

### PR DESCRIPTION
**Pull request by shahrozahmed1**
Fixes #15170 

Reduces the number of violations in the following files significantly:
- [x] `SANS/ISISCommandInterface.py`
- [x]  `reduction_gui/widgets/sans/stitcher.py`
- [x]  `reduction_gui/reduction/scripter.py`
- [x]  `scripts/SANS/isis_reducer.py`
- [x]  `reduction/instruments/sans/sans_reduction_steps.py`

**To Check**
- The functionalities of the files has not been affected so all the related tests should pass
- Code within the files have been cleaned up and reformatted according to `PEP-8`/`Pylint`
- Reduces around **148** `pylint` violations in total

Once merged, reset the jenkins pylint violations threshold

### CODE REVIEW

File: scripts/Interface/reduction_gui/reduction/scripter.py [hunk 1]
```diff
@@ -314,10 +319,10 @@ def __init__(self, name="", facility=""):
         if HAS_MANTID:
             config = ConfigService.Instance()
             try:
-                head, tail = os.path.split(config.getUserFilename())
+                head, _tail = os.path.split(config.getUserFilename())
                 if os.path.isdir(head):
                     self._output_directory = head
-            except:
+            except (StopIteration, StandardError, Warning):
```
Comment (picatess) [inline scripts/Interface/reduction_gui/reduction/scripter.py]: Are these all exception types which can occur?  Does pycharm display all possible exception types? Would be really usefull for me to know that.

Comment (shahrozahmed1) [inline scripts/Interface/reduction_gui/reduction/scripter.py]: @picatess Yes these are the three sub-trees to exception, however pylint doesn't seem to complain about writing them separately and yet covers all. Maybe this link could be useful: https://docs.python.org/2/library/exceptions.html#exception-hierarchy

Comment (picatess) [inline scripts/Interface/reduction_gui/reduction/scripter.py]: In that case I am not sure what the benefit is over disabling the pylint warning to be honest.

**Comment by picatess** [comment]
* Reran the system tests. Everything passed.
* Code review. Changes look good to me and are a big improvement in terms of code quality.
* Appears to be a reduction of 98 warnings
Happy to merge.

**Pull request merged**